rbc/typesystem.py

```diff
@@ -1109,6 +1109,8 @@ def fromnumba(cls, t):
         if isinstance(t, nb.types.misc.RawPointer):
             if t == nb.types.voidptr:
                 return cls(cls(), '*')
+        if isinstance(t, nb.types.StringLiteral):
+            return cls.fromstring(t.literal_value)
```

Comment:
This does not make sense:
```
>>> import numba
>>> import rbc.typesystem as ts
>>> ti=ts.TargetInfo('cpu')
>>> s=numba.types.StringLiteral('abc')
>>> s
Literal[str](abc)
>>> with ti: ts.Type.fromnumba(s)
... 
Type('abc',)
```
that is, a numba string literal becomes a type that name is the value of the string literal.

The `fromnumba` method should be applied to numba types only. For constructing Type from a value, one should use `Type.fromvalue` method.

So:
```suggestion
```